src/envoy/mixer/http_control.cc

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/mixer/http_control.h"
+#include "common/common/utility.h"
+#include "common/http/utility.h"
+
+using ::google::protobuf::util::Status;
+using ::istio::mixer_client::Attributes;
+using ::istio::mixer_client::DoneFunc;
+
+namespace Http {
+namespace Mixer {
+namespace {
+
+const std::string kProxyPeerID = "Istio/Proxy";
+const std::string kEnvNameTargetService = "TARGET_SERVICE";
+
+// Define lower case string for X-Forwarded-Host.
+const LowerCaseString kHeaderNameXFH("x-forwarded-host", false);
+
+// Define attribute names
+const std::string kAttrNamePeerId = "peerId";
+const std::string kAttrNameURL = "url";
+const std::string kAttrNameHttpMethod = "httpMethod";
```

Comment:
add x-request-id as an attribute since it's very useful for understanding dataflow.